web/src/compose_validate.ts

```diff
@@ -746,21 +747,30 @@ export function check_overflow_text(): number {
         set_message_too_long(false);
     } else {
         $indicator.text("");
-        $("textarea#compose-textarea").removeClass("over_limit");
+        $textarea.removeClass("over_limit");
 
         set_message_too_long(false);
     }
 
     return text.length;
+};
+
+export function rewire_check_overflow_text(value: typeof check_overflow_text): void {
+    check_overflow_text = value;
 }
 
-export function validate_message_length(): boolean {
-    if (compose_state.message_content().length > realm.max_message_length) {
-        $("textarea#compose-textarea").addClass("flash");
-        setTimeout(() => $("textarea#compose-textarea").removeClass("flash"), 1500);
+export let validate_message_length = ($elem: JQuery): boolean => {
+    const $textarea = $elem.find<HTMLTextAreaElement>(".message-textarea");
```

Comment:
The change from using `compose_state.message_content` means you're no longer trimming leading whitespace in doing this calculation. Technically that makes the calculation incorrect, in that the server will accept a message over the limit. So probably this should do `.trimEnd()` after doing `.val()`.